Self-enrollment / publishing release post (#105)
PrairieLearn/marketing: src/pages/about/blog/self-enrollment-overhaul/index.mdx

import {
  BlogMarkdownLayout,
  BlogImage,
  BlogCalloutBox,
} from "../../../../components/BlogMarkdownLayout";

import acceptInvitationHomepage from "./accept_invitation_homepage.png";
import inviteStudent from "./invite_student.png";
import selfEnrollmentSettingsFull from "./self_enrollment_settings_full.png";
import enterEnrollmentCodeHomepage from "./enter_enrollment_code_homepage.png";
import studentDetailsPage from "./student_details_page.png";
import scheduledPublish from "./scheduled_publish.png";
import addExtension from "./add_extension.png";
import releaseCover from "./release_cover.png";

export const meta = {
  title: "Overhauled enrollment management",
  date: "2026-01-01",
  author: "Peter Stenger",
  tags: ["Release"],
  introImage: releaseCover,
};

> The new self-enrollment and publishing systems are here! They bring many improvements, including the ability to manage everything in the UI (no need to edit JSON) and fine-grained self-enrollment controls.

Reviewer: Just to stop people from freaking out:
```suggestion
The new self-enrollment and publishing systems are here! They bring many improvements, including fine-grained self-enrollment controls and the ability to manage everything in the UI (JSON editing is still available, but it's no longer required).
```

<BlogCalloutBox variant="info">
  **Key Takeaways:**

1. Your existing course instances don't need any changes.
2. Going forward, how students can access your course instance is changing.
   You can either give them a link/code to enroll (see the [self-enrollment section](#self-enrollment)), or invite them yourself (see the [invitations section](#invitations)).
   There is no longer an enrollment page for students to find courses.

</BlogCalloutBox>

Let's get into all the new features!

### Enrollment management

You should now use the "Students" tab to manage enrollments. You no longer need to manage enrollments or access in JSON. There are two

> main ways to grant students access to your course:

Reviewer: ```suggestion
main ways to grant students access to your course instance:
```

2. Manually invite students to your course

TODO Students page image

### Self-enrollment

Self-enrollment allows students to enroll in a course instance as long as they have a code or link to join the course.
You can further limit self-enrollment to specific institutions or by setting a cutoff date for self-enrollment.

<BlogImage
  src={selfEnrollmentSettingsFull}
  alt="Overview of self-enrollment settings"
  caption="The full self-enrollment settings panel with all available options"
  full
/>

#### Self-enrollment codes

If you want to allow students to input a code to join your course, you should check "Use enrollment code for self-enrollment".
If they access any link to your course, they will be prompted to enter the code.

<BlogImage
  src={enterEnrollmentCodeHomepage}
  alt="Enter enrollment code on homepage"
  caption="Students can enter an enrollment code from the homepage to join a course instance"
  full
/>

### Invitations

You can also explicitly invite students to your course. These invitations will show up on the homepage for students to accept.
If you want full control over everyone in your course, consider disabling self-enrollment completely.

> You can invite a student to your course instance by clicking the "Invite" button on the "Students" tab.

Reviewer: I think we should maintain focus on explaining the features, not writing a user manual.
```suggestion
```

When preparing for a new term, you can set up your course instance to be published in the future. On that date,
the course instance will switch to published.

<BlogImage
  src={scheduledPublish}
  alt="Scheduled publishing"
  caption="You can schedule a course instance to be published automatically on a future date"
/>

Copying or creating a new course instance in the UI will automatically migrate you to the modern publishing system.
If you prefer to manage your course instance via JSON files, you can follow [the steps in the documentation](https://prairielearn.readthedocs.io/en/latest/courseInstance/#migrating-from-allowaccess) to migrate your course.

### Publishing extensions for individual students

You can grant an extension to give specific students access to the course instance beyond the original publish date. If a student has multiple extensions, the latest one takes effect.
These extensions are not written to the course instance's JSON file.

<BlogImage
  src={addExtension}
  alt="Add extension"
  caption="Granting an extension to specific students directly through the UI"
/>

### Other changes

- Students now manage their enrollments via the homepage.

> - The dedicated enrollment page that listed all course instances is going away.

Reviewer: ```suggestion
- The dedicated enrollment page that listed all course instances has been removed.
```
